src/main/java/CorgiManager/CorgiManager.java

```diff
@@ -0,0 +1,262 @@
+    //create a hashmap to map commands to respective methods
+    // CommandHandler is a functional interface to handle throwing exception for lambda function of COMMANDS hashtable
+    public static final Map<String, CommandHandler> COMMANDS = new HashMap<>();
+
+    // initialize commands
+    static {
+        COMMANDS.put("bye", input -> corgiPrint("Bye. Hope to see you again soon!"));
+        COMMANDS.put("list", input -> listTask());
+        COMMANDS.put("todo", input -> {
+            if (input.length < 2) {
+                throw new MissingArgumentException("Missing argument for 'todo' command");
+            }
+            addTask(input[1]);
+        });
+        COMMANDS.put("deadline", input -> {
+            if (input.length < 2) {
+                throw new MissingArgumentException("Missing argument for 'deadline' command");
+            }
+            addDeadline(input[1]);
+        });
+        COMMANDS.put("event", input -> {
+            if (input.length < 2) {
+                throw new MissingArgumentException("Missing argument for 'event' command");
+            }
+            addEvent(input[1]);
+        });
+        COMMANDS.put("delete", input -> {
+            if (input.length < 2) {
+                throw new MissingArgumentException("Missing argument for 'delete' command");
+            }
+            removeTask(input[1]);
+        });
+        COMMANDS.put("mark", input -> {
+            if (input.length < 2) {
+                throw new MissingArgumentException("Missing argument for 'mark' command");
+            }
+            markTask(input[1]);
+        });
+        COMMANDS.put("unmark", input -> {
+            if (input.length < 2) {
+                throw new MissingArgumentException("Missing argument for 'unmark' command");
+            }
+            unmarkTask(input[1]);
+        });
+    }
+
+    //tasks is used to store Task objects created by user
+    public static ArrayList<Task> tasks = new ArrayList<>();
+
+    public static void main(String[] args) {
+
+        System.out.println("Hello! I'm Corgi");
+        System.out.println("What can I do for you?");
+        System.out.println(partition);
+
+        loadTasks();
+
+        Scanner scanner = new Scanner(System.in);
+
+        while (true) {
+            String input = scanner.nextLine().trim();
+            String[] command = input.split(" ", 2);
+
+            //Handle checked exceptions
+            try {
+                if (COMMANDS.containsKey(command[0])) {
+                    //handle method is defined in CommandHandler
+                    COMMANDS.get(command[0]).handle(command);
+                    if (command[0].equalsIgnoreCase("bye")) {
+                        break;
+                    }
+                } else {
+                    throw new InvalidCommandException("Command does not exist!!");
+                }
+            } catch (MissingArgumentException e) {
+                //System.err.println might be better
```

Comment:
The comments should be for the reader. This comment seems to be for your own reference.